### PR TITLE
Refactor ci-publish.sh

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

**Comment by MatthewPhinney** [comment]
LGTM

File: bin/ci-publish.sh [hunk 1]
```diff
@@ -2,35 +2,44 @@
...
 
-checkVersion() {
-  local gitTag
-  gitTag=$(git describe --tags)
+getLatestTag() {
+  git describe --abbrev=0 --tags
+}
+
+getCommitHash() {
+  git rev-list -n 1 $1
```
Comment (lumberj) [inline bin/ci-publish.sh]: What is $1 here?

**Comment by lumberj** [comment]
LGTM

File: bin/ci-publish.sh (hunk 1 above)
Comment (clark800) [inline bin/ci-publish.sh]: First parameter to the function

File: bin/ci-publish.sh [hunk 2]
```diff
@@ -2,35 +2,44 @@
...
+}
+
+checkHeadIsLatestTag() {
+  if [ "$(getCommitHash HEAD)" != "$(getCommitHash $(getLatestTag))" ]; then
```
Comment (lumberj) [inline bin/ci-publish.sh]: Won't these always be equal?

Comment (lumberj) [inline bin/ci-publish.sh]:
```
Projects/five-bells-shared [master] » git --no-pager log -n 1
Thu Mar 24 14:29:06 2016 -0700 bdf825c (HEAD -> master, upstream/master) Merge pull request #123 from lumberj/publish-task  [Alan Cohen]%                                            

Projects/five-bells-shared [master] » g --no-pager show $(git rev-list -n 1 $(git describe))
Thu Mar 24 14:29:06 2016 -0700 bdf825c (HEAD -> master, upstream/master) Merge pull request #123 from lumberj/publish-task  [Alan Cohen]

Projects/five-bells-shared [master] » g --no-pager show $(git rev-list -n 1 HEAD)
Thu Mar 24 14:29:06 2016 -0700 bdf825c (HEAD -> master, upstream/master) Merge pull request #123 from lumberj/publish-task  [Alan Cohen]
```

Comment (lumberj) [inline bin/ci-publish.sh]: I see, it's the `abbrev=0`

```
Projects/five-bells-shared [master] » git --no-pager describe --abbrev=0 --tags
v12.3.0
Projects/five-bells-shared [master] » git --no-pager describe --tags
v12.3.0-2-gbdf825c
```

Comment (clark800) [inline bin/ci-publish.sh]: Yeah, without `abbrev=0` it's not the actual tag name, which is why the previous code was confusing

**Pull request merged**